primitives/src/merkle_tree/universal_merkle_tree.rs

```diff
@@ -285,6 +299,30 @@ mod mt_tests {
             .unwrap()
             .is_ok());
         }
+        for i in 0..10 {
+            mt.update_with(F::from(i as u64), |elem| match elem {
+                Some(elem) => Some(*elem),
+                None => Some(F::from(i as u64)),
+            })
+            .unwrap();
+        }
+        assert_eq!(mt.num_leaves(), 10);
+        // test lookup at index 7
+        let (val, proof) = mt.universal_lookup(F::from(7u64)).expect_ok().unwrap();
+        assert_eq!(val, &F::from(7u64));
```

Comment:
why don't we test all pos that are updated?
also how are sure they are new value? I thought leaf at pos `i` was originally also initiated as `F::from(i)`?

Reply:
Index 7 is not updated previously